site/src/components/Workspace/Workspace.tsx

```diff
@@ -131,6 +132,40 @@ export const Workspace: FC<React.PropsWithChildren<WorkspaceProps>> = ({
   if (template !== undefined) {
     transitionStats = ActiveTransition(template, workspace)
   }
+
+  const [showAlertPendingInQueue, setShowAlertPendingInQueue] = useState(false)
+  const now = dayjs()
+  useEffect(() => {
+    if (
+      workspace.latest_build.status === "pending" &&
+      workspace.latest_build.job.queue_size > 0 &&
+      dayjs(workspace.latest_build.created_at).isBefore(
+        now.subtract(5, "seconds"),
+      )
+    ) {
+      setShowAlertPendingInQueue(true)
+      return
+    }
+
+    if (
+      workspace.latest_build.status === "pending" &&
+      workspace.latest_build.job.queue_size > 0
```

Comment:
I noticed we're not setting `setShowAlertPendingInQueue(false)` anywhere? Perhaps this condition could be inverted and placed at the top with an false + early return. Should simplify the other checks as well.

Reply:
Fixed